### ISSUE TITLE
`terraform_unused_required_providers` does not track providers passed to modules

### ISSUE THREAD

**Issue by bendrucker** [terraform-linters/tflint#1189]
The following is a valid usage of a required provider. Currently, TFLint reports `null` as unused.

```hcl
terraform {
  required_providers {
    null = { source = "hashicorp/null" }
  }
}

module "foo" {
  providers = {
    null = null
  }
}
```

This is particularly common with [provider aliases](https://www.terraform.io/docs/language/modules/develop/providers.html#provider-aliases-within-modules). A module might require multiple copies of the same provider all of which have aliases. Different provider instances might be passed to different module instances. As long as any alias of the required provider is used, it should count as used.

**Comment by sergei-ivanov** [comment]
Yes please! I am currently going through a rather painful migration of 200+ modules to Terraform 1.x and I am using TFlint as one of the tools to validate the migration. In one of the previous iterations we followed TFlint suggestions to remove the "unused" provider declarations, which worked OK with Terraform 0.14, but now I am having to add those blocks back and tag them with `tflint-ignore`.
